File: payola/web/public/javascripts/cz/payola/web/client/graph/model/Edge.js
goog.provide('cz.payola.web.client.graph.model.Edge');
cz.payola.web.client.graph.model.Edge = function(uri, from, to) {
var self = this;
self.uri = uri;
self.from = from;
self.to = to;
};
cz.payola.web.client.graph.model.Edge.prototype.metaClass_ = new s2js.MetaClass('cz.payola.web.client.graph.model.Edge', []);

File: payola/web/client/src/main/scala/cz/payola/web/client/views/graph/sigma/Edge.scala
package cz.payola.web.client.views.graph.sigma

import s2js.adapters.js.sigma
import cz.payola.domain.entities.settings.OntologyCustomization
import s2js.compiler.javascript

class Edge extends sigma.Edge {

    def updateProperties(newConfiguration: Any) {
        saveCurrentProperties()
        this.color = newConfiguration.asInstanceOf
            [OntologyCustomization#ClassCustomizationType].fillColor //sigma supports only color
    }

    @javascript("self.attr['real_color'] = self.color;")
    def saveCurrentProperties() {}

    @javascript("self.color = self.attr['real_color'];")
    def restartProperties() {}
}
